documentation/modules/metrics/proc-metrics-kafka-deploy-options.adoc

```diff
@@ -150,12 +151,13 @@ spec:
 <3> A regular expression to specify the topics to include in the metrics.
 <4> A regular expression to specify the consumer groups to exclude in the metrics.
 <5> A regular expression to specify the topics to exclude in the metrics.
-<6> CPU and memory resources to reserve.
-<7> Logging configuration, to log messages with a given severity (debug, info, warn, error, fatal) or above.
-<8> Boolean to enable Sarama logging, a Go client library used by Kafka Exporter.
-<9> Customization of deployment templates and pods.
-<10> Healthcheck readiness probes.
-<11> Healthcheck liveness probes.
+<6> By default, metrics are collected for all consumers regardless of their connection status. Setting `showAllOffsets` to `false` stops collecting metrics on disconnected consumers, preventing potentially misleading metrics from persisting and impairing the ability to detect lag in active consumers.
```

Comment:
Maybe we should stick to what it does?

```suggestion
<6> By default, metrics are collected for all consumers regardless of their connection status. Setting `showAllOffsets` to `false` stops collecting metrics on disconnected consumers.
```

Because I'm not sure I would say it is _potentially misleading metrics from persisting and impairing the ability to detect lag in active consumers_. I would instead warn about setting it to false because it _hides the problems with the growing lag of currently disconnected consumers_.

Reply:
I thought the problem was that it was hardcoded to true? In the related [issue](https://github.com/strimzi/strimzi-kafka-operator/issues/9231), the disconnected consumers skew the metrics.

Reply:
I think that is a subject to opinion as to whether it being hardcoded to true was a problem or not. For someone apparently yes, but not for others. I would probably stick with what it does and let everyone decide.